kusd/validator/states.go

```diff
@@ -31,54 +31,22 @@ func (val *Validator) notLoggedInState() stateFn {
 	isGenesis, err := val.network.IsGenesisVoter(&bind.CallOpts{}, val.account.Address)
 	if err != nil {
 		log.Crit("Failed to verify the voter information", "err", err)
-		return nil
 	}
 
 	// @NOTE (rgeraldes) - sync was already done at this point and by default the investors will be
 	// part of the initial set of validators - no need to make a deposit if the block number is 0
 	// since these validators will be marked as voters from the start
-	if !isGenesis || (isGenesis && val.chain.CurrentBlock().NumberU64() > 0) {
-		// Subscribe events from blockchain
-		chainHeadCh := make(chan core.ChainHeadEvent)
-		chainHeadSub := val.chain.SubscribeChainHeadEvent(chainHeadCh)
-		defer chainHeadSub.Unsubscribe()
-
+	if isFirstBlock(isGenesis, val) {
```

Comment:
isFirstBlock should be used only for this part of the condition - val.chain.CurrentBlock().NumberU64() > 0
Genesis validators are election voters (consensus) that are included from the block 0 (genesis block) - could be an assert like isVoter.

Reply:
ha!